rskj-core/src/test/java/co/rsk/core/types/bytes/BytesSliceTest.java

```diff
@@ -74,4 +113,44 @@ void testEmptySlice() {
         assertEquals(0, actualResult.length());
         assertArrayEquals(expectedResult, actualResult.copyArray());
     }
+
+    private static void checkArraycopy(Functions.Action5<Object, Integer, Object, Integer, Integer> fun) {
+        byte[] dest = new byte[3];
+        byte[] origin = new byte[]{1,2,3,4,5};
+
+        assertThrows(NullPointerException.class, () -> fun.apply(origin, 0, null, 0, 3));
+
+        assertThrows(IndexOutOfBoundsException.class, () -> fun.apply(origin, -1, dest, 0, 3));
+        assertThrows(IndexOutOfBoundsException.class, () -> fun.apply(origin, 0, dest, -1, 3));
+        assertThrows(IndexOutOfBoundsException.class, () -> fun.apply(origin, 0, dest, 0, -1));
+        assertThrows(IndexOutOfBoundsException.class, () -> fun.apply(origin, 0, dest, 0, 4));
+        assertThrows(IndexOutOfBoundsException.class, () -> fun.apply(origin, 1, dest, 0, 3));
+        assertThrows(IndexOutOfBoundsException.class, () -> fun.apply(origin, 0, dest, 1, 3));
+
+        assertArrayEquals(new byte[3], dest); // yet unmodified
+
+        fun.apply(origin, 0, dest, 0, 3);
+        assertArrayEquals(new byte[]{2,3,4}, dest);
+
+        byte[] dest2 = new byte[3];
+        fun.apply(origin, 1, dest2, 1, 1);
+        assertArrayEquals(new byte[]{0,3,0}, dest2);
+    }
+
+    private static <T> void checkCopyOfRange(Functions.Function3<T, Integer, Integer, byte[]> fun,
```

Comment:
suggestion:

The suggestions to add some local variables with meaningful names applies here as well. I think it would improve the readability from the test.